R/get_population.R

```diff
@@ -0,0 +1,140 @@
+#' have data for.
+#' @param gender both (only total), female, male, all (female, male, and total)
+#' @param years.start Life table data is available for a range of years. This is 
+#' the first year from which life table data will be returned. Minimum is 1950.
+#' @param years.end The end of the period of life table data that is returned. Maximum is 2100. 
+#' @param ages.start Life table data starts at age 0 (default), and is bounded in 5 year age
+#' groups.If this age is not available, will return 
+#' a table from the lowest bound of the age group older than the age specified.
+#' @param ages.end This is the last age of the specified life table. Since the life table is in groups, will 
+#' return a life table where the oldest age group includes the ages.end value. 
+#' @param age.groups TRUE/FALSE. Defaults to TRUE, if TRUE returns population by 
+#' 5 year age bands, which is available from 1950-2100.
+#' If FALSE, returns population estimates by single age, available range is from 1950-2019.
+#' 
+#' @keywords UN, demography, download, population, life table, deathrates, mx, ax, dx, gender, age groups
+#' @return Data frame of life table data.
+#' @export
+#' 
+#' @import vroom
+#' @import countrycode
+#' @import tidyr
+#' @import data.table
+#' @import stringr
+#' 
+#' @examples
+#' get_population(country="Norway")
+get_population <- function(country="all", 
+                          gender="all", 
+                          years.start=2000, 
+                          years.end=NA, 
+                          ages.start=0,
+                          ages.end=NA,
+                          age.groups=TRUE){
+  
+  stopifnot("Country must be a character." = is.character(country))
+  stopifnot("Gender must be a character."=is.character(gender))
+  stopifnot("Years.start must be numeric"=is.numeric(years.start))
+  stopifnot("Ages.start must be numeric"=is.numeric(ages.start))
+  stopifnot("Age.groups must be logical"=is.logical(age.groups))
+  
+  if(age.groups==TRUE){
+    url<-"https://population.un.org/wpp/Download/Files/1_Indicators%20(Standard)/CSV_FILES/WPP2019_PopulationByAgeSex_Medium.csv"
+  } else {
+    url<-"https://population.un.org/wpp/Download/Files/1_Indicators%20(Standard)/CSV_FILES/WPP2019_PopulationBySingleAgeSex_1950-2019.csv"
+  }
+
+  DT = vroom::vroom(url)
+  
+  # check format of country input and reformat to country name
+  
+  if(country != "all") {
+    # convert iso3c codes
+    if(country %in% countrycode::codelist$iso3c){
+      country <- countrycode::countrycode(country, origin="iso3c", destination="country.name")
+    } 
+    # filter by country - has tolower in case someone misspelled something
+    if(tolower(country) %in% tolower(DT$Location)){
+      DT = DT[tolower(DT$Location) == tolower(country),]
+    }
+  }
+
+  # recode gender so that it's the same as the life table data
+  names(DT)[names(DT) == "PopMale"] <- "Male"
```

Comment:
This might be just how the data comes, but anywhere we don't need capitalisation its worth avoiding (e.g. "Male" or "Population" etc).
It adds an uncesscary key-stroke each time so good to ditch where reasonable